tap_pardot/client.py

```diff
@@ -145,7 +227,7 @@ def describe(self, endpoint, **kwargs):
         jitter=None,
     )
     def _fetch(self, method, endpoint, format_params, **kwargs):
-        base_formatting = [endpoint, self.api_version]
+        base_formatting = [endpoint, '{}']
```

Comment:
This looks really weird, why are you calling format here, putting a thing that will be later formatted again with the api version?

Reply:
With old auth, the appropriate api version is determined on login. As far as we know, oauth doesn't provide a way to figure out what api version to use. Our solution is to guess that the api is version 4 in the init, then change the api version to 3 if we catch an 89 error in requests. Passing '{}' lets us format and add the api version after we get an 89.

Reply:
`[endpoint, '{api_version}'] is an option and then in the second format use the keyword .format method: https://realpython.com/python-formatted-output/